src/core/smtp/MCSMTPSession.cpp

```diff
@@ -672,17 +672,22 @@ void SMTPSession::sendMessage(Address * from, Array * recipients, Data * message
         goto err;
     }
     else if (r != MAILSMTP_NO_ERROR) {
-        if ((responseCode == 550) && (response != NULL) && (response->hasPrefix(MCSTR("5.3.4")))) {
-            * pError = ErrorNeedsConnectToWebmail;
-            goto err;
-        }
-        else {
-            * pError = ErrorSendMessage;
-            MC_SAFE_REPLACE_COPY(String, mLastSMTPResponse, response);
-            mLastLibetpanError = r;
-            mLastSMTPResponseCode = responseCode;
-            goto err;
+        if ((responseCode == 550) && (response != NULL)) {
+            if (response->hasPrefix(MCSTR("5.3.4"))) {
```

Comment:
According to https://tools.ietf.org/html/rfc3463:

```
3.8 Security or Policy Status

  X.7.1   Delivery not authorized, message refused

         The sender is not authorized to send to the destination.  This
         can be the result of per-host or per-recipient filtering.  This
         memo does not discuss the merits of any such filtering, but
         provides a mechanism to report such.  This is useful only as a
         permanent error.
```

ErrorSendMessageNotAllowed sounds like a good error code to me.
Let me know if you have some concern.